cmd/assetsvc/postgresql_utils.go

```diff
@@ -36,32 +38,137 @@ func newPGManager(config datastore.Config) (assetManager, error) {
 }
 
 func (m *postgresAssetManager) getChart(chartID string) (models.Chart, error) {
-	// TODO: Implement this!
-	return models.Chart{}, fmt.Errorf("Unimplemented")
+	var chart models.ChartIconString
+	err := m.QueryOne(&chart, fmt.Sprintf("SELECT info FROM %s WHERE info ->> 'ID' = $1", dbutils.ChartTable), chartID)
+	if err != nil {
+		return models.Chart{}, err
+	}
+
+	// TODO(andresmgot): Store raw_icon as a byte array
+	icon, err := base64.StdEncoding.DecodeString(chart.RawIcon)
+	if err != nil {
+		return models.Chart{}, err
+	}
+	return models.Chart{
+		ID:              chart.ID,
+		Name:            chart.Name,
+		Repo:            chart.Repo,
+		Description:     chart.Description,
+		Home:            chart.Home,
+		Keywords:        chart.Keywords,
+		Maintainers:     chart.Maintainers,
+		Sources:         chart.Sources,
+		Icon:            chart.Icon,
+		RawIcon:         icon,
+		IconContentType: chart.IconContentType,
+		ChartVersions:   chart.ChartVersions,
+	}, nil
 }
 
 func (m *postgresAssetManager) getChartVersion(chartID, version string) (models.Chart, error) {
-	// TODO: Implement this!
-	return models.Chart{}, fmt.Errorf("Unimplemented")
+	var chart models.Chart
+	err := m.QueryOne(&chart, fmt.Sprintf("SELECT info FROM %s WHERE info ->> 'ID' = $1", dbutils.ChartTable), chartID)
+	if err != nil {
+		return models.Chart{}, err
+	}
+	for _, c := range chart.ChartVersions {
+		if c.Version == version {
+			chart.ChartVersions = []models.ChartVersion{c}
+			break
+		}
+	}
+	return chart, nil
 }
 
 func (m *postgresAssetManager) getChartFiles(filesID string) (models.ChartFiles, error) {
-	// TODO: Implement this!
-	return models.ChartFiles{}, fmt.Errorf("Unimplemented")
+	var chartFiles models.ChartFiles
+	err := m.QueryOne(&chartFiles, fmt.Sprintf("SELECT info FROM %s WHERE chart_files_id = $1", dbutils.ChartFilesTable), filesID)
+	if err != nil {
+		return models.ChartFiles{}, err
+	}
+	return chartFiles, nil
+}
+
+func containsVersionAndAppVersion(chartVersions []models.ChartVersion, version, appVersion string) (models.ChartVersion, bool) {
+	for _, ch := range chartVersions {
+		if ch.Version == version && ch.AppVersion == appVersion {
+			return ch, true
+		}
+	}
+	return models.ChartVersion{}, false
 }
 
 func (m *postgresAssetManager) getChartsWithFiltes(name, version, appVersion string) ([]*models.Chart, error) {
-	// TODO: Implement this!
-	return nil, fmt.Errorf("Unimplemented")
+	charts, err := m.QueryAllCharts(fmt.Sprintf("SELECT info FROM %s WHERE info ->> 'name' = $1", dbutils.ChartTable), name)
+	if err != nil {
+		return nil, err
+	}
+	result := []*models.Chart{}
+	for _, c := range charts {
+		if cv, found := containsVersionAndAppVersion(c.ChartVersions, version, appVersion); found {
+			// Return only the interesting ChartVersion
+			c.ChartVersions = []models.ChartVersion{cv}
+			result = append(result, c)
+		}
+	}
+	return result, nil
 }
 
+// NOTE: searchCharts is not currently being used in Kubeapps
```

Comment:
If we implement pagination (https://github.com/kubeapps/kubeapps/issues/1010) we would need to use this code to filter charts in the server side. I don't think that's going to happen anytime soon though. I don't mind removing the code, in any case it's not exactly the same functionality than the one for Mongo (because of the TODOs I set).